tendrl/node_agent/message/logger.py

```diff
@@ -0,0 +1,93 @@
+from tendrl.commons.message import Message
+
+
+LOG = logging.getLogger(__name__)
+
+
+class Logger(object):
+    def __init__(self, message):
+        self.message = message
+        self.etcd_orm = tendrl_ns.etcd_orm
+        self.push_messages()
+        if "job_id" in message.payload:
+            """ If job_id is present then
+
+            it is considered as operation
+            """
+            self._logger(self.push_operation())
+        else:
+            self._logger(self.message.payload["message"])
+
+    def push_operation(self):
+        self.etcd_orm.client.write(
+            self.message.payload["job_id"],
+            Message.to_json(self.message),
+            append=True)
+        log_message = ("%s:%s") % (
+            self.message.payload["job_id"],
+            self.message.payload["message"])
+        return log_message
+
+    def push_messages(self):
+        if self.message.priority not in [
+            Message.priorities.INFO, Message.priorities.DEBUG]:
+            # Stroring messages cluster wise
+            if "cluster_id" in self.message.payload:
+                self.etcd_orm.client.write(
+                    ("clusters/%s/messages/%s") % (
+                        self.message.payload["cluster_id"],
+                        self.message.message_id),
+                    Message.to_json(self.message)
+                )
+            # storing messages node wise
+            else:
+                self.etcd_orm.client.write(
+                    ("nodes/%s/messages/%s") % (
+                        self.message.node_id,
+                        self.message.message_id),
+                    Message.to_json(self.message)
+                )
+            # storing messages global wise
+            self.etcd_orm.client.write(
+                ("messages/%s") % (
+                    self.message.message_id),
+                Message.to_json(self.message)
+            )
+
+    def _logger(self, log_message):
+        # Invalid message
+        if isinstance(log_message, Message):
+            log_message = Message.to_json(log_message)
+        message = ("%s - %s - %s:%s - %s - %s - %s") % (
+            self.message.timestamp,
+            self.message.publisher,
+            self.message.caller["filename"],
+            self.message.caller["line_no"],
+            self.message.caller["function"],
+            self.message.priority.upper(),
+            log_message
+        )
+        try:
+            method = getattr(self, self.message.priority)
+        except AttributeError:
+            raise NotImplementedError(self.message.priority)
+        method(message)
+
+    def info(self, message):
+        LOG.info(message)
+
+    def error(self, message):
+        LOG.error(message)
+
+    def debug(self, message):
+        LOG.debug(message)
+
+    def warning(self, message):
+        LOG.warning(message)
+
+    def critical(self, message):
+        LOG.critical(message)
+
+    def notice(self, message):
```

Comment:
I see from line 72, that this func name should match priority I think case wise and also exact name should be the func here. In which case is it NOTIF or notice? Applies to all other funcs above as well. Also if this all that needs to be done is there a better way of doing it? Probably have a map from log levels of tendrl to log levels of logger and use it in getattr directly on LOG object instead of on self? Just a thought.